src/fs/metadata_log.cc

```diff
@@ -136,37 +183,48 @@ void metadata_log::memory_only_delete_inode(inode_t inode) {
         [](const inode_info::directory& dir) {
             assert(dir.entries.empty());
         },
-        [](const inode_info::file&) {
-            // TODO: for compaction: update used inode_data_vec
+        [&](inode_info::file& file) {
+            cut_out_data_range(file, {
+                0,
+                std::numeric_limits<decltype(file_range::end)>::max()
+            });
+            _rewrite_log_size -= ondisk_entry_size<ondisk_truncate>();
         }
     }, it->second.contents);
 
     _inodes.erase(it);
+    _rewrite_log_size -= ondisk_entry_size<ondisk_create_inode>();
 }
 
 void metadata_log::memory_only_small_write(inode_t inode, file_offset_t file_offset, temporary_buffer<uint8_t> data) {
     inode_data_vec data_vec = {
         {file_offset, file_offset + data.size()},
         inode_data_vec::in_mem_data {std::move(data)}
     };
-
     auto it = _inodes.find(inode);
     assert(it != _inodes.end());
     assert(it->second.is_file());
     write_update(it->second.get_file(), std::move(data_vec));
+    _rewrite_log_size += ondisk_entry_size<ondisk_small_write_header>(data.size());
 }
 
 void metadata_log::memory_only_disk_write(inode_t inode, file_offset_t file_offset, disk_offset_t disk_offset,
         size_t write_len) {
+    assert(write_len <= _cluster_size);
     inode_data_vec data_vec = {
         {file_offset, file_offset + write_len},
         inode_data_vec::on_disk_data {disk_offset}
     };
-
     auto it = _inodes.find(inode);
     assert(it != _inodes.end());
     assert(it->second.is_file());
     write_update(it->second.get_file(), std::move(data_vec));
+    _rewrite_log_size += (write_len == _cluster_size ?
+        ondisk_entry_size<ondisk_large_write>() :
+        ondisk_entry_size<ondisk_medium_write>());
+    cluster_id_t cluster_id = offset_to_cluster_id(disk_offset, _cluster_size);
+    _writable_data_clusters[cluster_id].add_data(disk_offset, inode, {file_offset, file_offset + write_len});
```

Comment:
```suggestion
    _writable_data_clusters.at(cluster_id).add_data(disk_offset, inode, {file_offset, file_offset + write_len});
```
Use at() (preferably i would like to have version of at() that instead of throwing an exception calls std::terminate -- asserts that the key exists). operator[] creates element in the map, so if the write is misused here, there will be no error.

Reply:
Please look at the comment about at() that is somewhere above.